### DISCUSSION TITLE
How do I pass the None value explicitly?

### DISCUSSION THREAD

**Discussion by salmiakki** [fastapi/typer#1340] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Optional

import typer


def main(
    limit: Optional[int] = typer.Option(11),
):
    print(limit)

if __name__ == "__main__":
    typer.run(main)
```

### Description

How do I pass the `None` value for an `Optional` argument that has a non-`None` default value?

### Operating System

macOS

### Typer Version

0.7.0

### Python Version

Python 3.9.13

**Comment by jonaslb** [comment]
Semantically `None` is used to communicate the absence of a specified value in typer. So you can't as far as I can tell set it explicitly or enable functionality to allow it.

My proposal is that you either use a special value (for example `-1` or `0`) to indicate the limit, or to create a special `--disable-limit` flag, like so:

```python
from typing import Optional

import typer

def main1(
    limit: Optional[int] = typer.Option(11, help="A limit value of 0 disables the limit.")
):
    if limit > 0:
        print(limit)
    else:
        print("unlimited")

def main2(
    limit: Optional[int] = 11,
    disable_limit: bool = typer.Option(False, "--disable-limit")
):
    if disable_limit:
        limit = None
    print(limit)

if __name__ == "__main__":
    typer.run(main1)

```

**Comment by kuykendall-ben** [comment, marked as answer]
Update here: with custom parsers, you should be able to implement this as of [typer 0.8.0](https://github.com/tiangolo/typer/releases/tag/0.8.0)!

```
from typing import Optional
import typer

def int_or_none(raw: str) -> Optional[int]:
    return None if raw == 'None' else int(raw)

def main(
    limit: Optional[int] = typer.Option(11, parser=int_or_none),
):
    print(limit)

if __name__ == "__main__":
    typer.run(main)
```

Usage:
```
$ /usr/bin/python3 main.py       
11
$ /usr/bin/python3 main.py --limit None
None
$ /usr/bin/python3 main.py --limit 3   
3
```

The name of the parser is even displayed in the help string:
```
$ /usr/bin/python3 main.py --help      
Usage: main.py [OPTIONS]

Options:
  --limit INT_OR_NONE  [default: 11]
  --help               Show this message and exit.

```
You could modify the `metavar` or `help` field of your `Option` for additional clarify.
